obs-studio-server/source/nodeobs_service.cpp

```diff
@@ -911,110 +893,230 @@ bool OBS_service::startStreaming(void)
 		obs_output_release(streamingOutput);
 
 	streamingOutput = obs_output_create(type, "simple_stream", nullptr, nullptr);
+	if (!streamingOutput)
+		return false;
+
 	connectOutputSignals();
 
-	updateAudioStreamingEncoder();
+	std::string currentOutputMode = config_get_string(ConfigManager::getInstance().getBasic(), "Output", "Mode");
+	bool        isSimpleMode      = currentOutputMode.compare("Simple") == 0;
+
+	updateStreamingEncoders(isSimpleMode);
 	updateService();
-	updateStreamSettings();
+	updateStreamingOutput(isSimpleMode);
+
+	obs_output_set_video_encoder(streamingOutput, videoStreamingEncoder);
+
+	if (isSimpleMode)
+		obs_output_set_audio_encoder(streamingOutput, audioSimpleStreamingEncoder, 0);
+	else
+		obs_output_set_audio_encoder(
+		    streamingOutput,
+		    audioAdvancedStreamingEncoder,
+		    config_get_int(ConfigManager::getInstance().getBasic(), "AdvOut", "TrackIndex") - 1);
 
 	isStreaming = true;
 	return obs_output_start(streamingOutput);
 }
 
-bool OBS_service::updateAudioStreamingEncoder()
+void OBS_service::updateAudioStreamingEncoder(bool isSimpleMode)
 {
-	std::string currentOutputMode = config_get_string(ConfigManager::getInstance().getBasic(), "Output", "Mode");
-	bool        advanced          = currentOutputMode.compare("Advanced") == 0;
-	std::string id;
-	const char* codec = NULL;
+	const char* codec = obs_output_get_supported_audio_codecs(streamingOutput);
 
-	if (!streamingOutput) {
-		createStreamingOutput();
-		codec = obs_output_get_supported_audio_codecs(streamingOutput);
-		obs_output_release(streamingOutput);
-		streamingOutput = nullptr;
-	} else {
-		codec = obs_output_get_supported_audio_codecs(streamingOutput);
+	if (!codec) {
+		return;
 	}
 
-	if (!codec) {
-		return false;
+	obs_encoder_t** enc = isSimpleMode ? &audioSimpleStreamingEncoder : &audioAdvancedStreamingEncoder;
+
+	if (*enc && obs_encoder_active(*enc))
+		return;
+
+	if (*enc && isSimpleMode) {
```

Comment:
If this condition works we have *enc released and set to null. 
In next condition "**(strcmp(codec, "aac") == 0 && isSimpleMode)**" we did not recreate this reference but use it 18 lines below in "obs_encoder_set_audio(*enc, obs_get_audio());"

And in another condition we check "**(strcmp(codec, "aac") == 0 && !isSimpleMode)**" and save to *enc another value. can it be a leak of encoder?